packages/entity/src/EntityQueryContext.ts

```diff
@@ -65,6 +79,17 @@ export class EntityTransactionalQueryContext extends EntityQueryContext {
     this.postCommitCallbacks.push(callback);
   }
 
+  public async runPreCommitCallbacksAsync(): Promise<void> {
```

Comment:
IMO it's worth reaffirming one key thing in the documentation: the pre-commit callbacks are run according to their order (should we document that the order is stable otherwise? I'm thinking no, since if you care about the order of callbacks, you should express that with `order`, not with the order in which they are registered).

Relatedly, do you think it is worth grouping callbacks by their order and running each group with Promise.all? (IMO: not that important right now, but may want to leave the API contract open to allow this, in which case we don't want to document that the callbacks are run serially -- just that they are run in order.)

Reply:
Good call. I think it's fine to not group them for now, but I'll update the documentation to indicate that it may be possible in the future.